### PR TITLE
Add envoy sample to demonstrate gRPC-JSON transcoding.

### PR DESCRIPTION
**Pull request by timburks**

### CODE REVIEW

File: envoy/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Run the Kiosk API with Google Cloud Endpoints
```
[the remark this answers is not in the document]
Comment (timburks) [reply envoy/README.md]: Thanks for catching that! Corrected.

File: envoy/README.md [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-
-1.  Rebuild the protos with API descriptors for envoy to consume.
+1.  Rebuild the protos with API descriptors for Envoy to consume.
 
     ```bash
     ./COMPILE_PROTOS.sh
     ```
 
-1.  Start the envoy proxy.
+1.  Start Envoy.
```
Comment (noahdietz) [inline envoy/README.md]: This should be `2.`

File: envoy/README.md [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
 
     ```bash
     envoy -c envoy.yaml
     ```
 
+1. Access the API with curl or a web browser. For example [this address](http://localhost:8081/v1/kiosks) calls the ListKiosks API method.
```
Comment (noahdietz) [inline envoy/README.md]: This should be `3.`

Comment (timburks) [reply envoy/README.md]: Markdown [automatically numbers lists](https://stackoverflow.com/questions/12145232/create-an-automatically-numbered-list). 

Verifiable in this case [here](https://github.com/timburks/kiosk/tree/master/envoy).

**Pull request merged**